tasks/post_to_catalog/post_to_catalog.py

```diff
@@ -197,26 +188,13 @@ def create_collection_sql():
     """
     )
 
-
-def create_provider_collection_xref_sql():
-    return text(
-        """
-    INSERT INTO provider_collection_xref
-            (provider_id, collection_id)
-    VALUES
-        (:provider_id, :collection_id)
-    ON CONFLICT DO NOTHING
-    """
-    )
-
-
 def create_granule_sql():
     return text(
         """
     INSERT INTO granules
-        (collection_id, cumulus_granule_id, execution_id, ingest_time, cumulus_create_time, last_update)
+        (provider_id, collection_id, cumulus_granule_id, execution_id, ingest_time, cumulus_create_time, last_update)
     VALUES
-        (:collection_id, :cumulus_granule_id, :execution_id, :ingest_time, :cumulus_create_time, :last_update)
+        (:provider_id, :collection_id, :cumulus_granule_id, :execution_id, :ingest_time, :cumulus_create_time, :last_update)
     ON CONFLICT (collection_id, cumulus_granule_id) DO UPDATE
         SET 
             execution_id=:execution_id, last_update=:last_update
```

Comment:
I wonder if we shouldn't also update the provider_id here along with the execution_id. Thoughts? @AndrewDorn  and @rizbi-hassan

Reply:
Ah, maybe. Don't know if Cumulus would throw a modified provider_id for a granule at us.

Reply:
added provider_id under `ON CONFLICT`

Reply:
That is not the correct place. The conflict here is based on a unique constraint that states the collection and granule id combination must be unique. Per my comment this would update the provider id value along with the execution id and date.

should be 

```
ON CONFLICT (collection_id, cumulus_granule_id) DO UPDATE
        SET 
            provider_id=:provider_id, execution_id=:execution_id, last_update=:last_update
  RETURNING id"""
```